slice/internal/controller/workload_controller.go

```
@@ -161,11 +163,28 @@ func (r *WorkloadReconciler) shouldFinalize(wl *kueue.Workload) bool {
 	return !wl.DeletionTimestamp.IsZero() || workload.IsFinished(wl) || workload.IsEvicted(wl) || !workload.IsActive(wl)
 }
 
-func (r *WorkloadReconciler) isRelevantWorkload(wl *kueue.Workload) bool {
-	return hasRelevantPodSet(wl.Spec.PodSets) &&
-		workload.HasQuotaReservation(wl) &&
-		wl.Status.Admission != nil &&
-		hasRelevantPodSetAssignment(wl.Status.Admission.PodSetAssignments)
+func (r *WorkloadReconciler) isRelevantWorkload(wl *kueue.Workload, log logr.Logger) bool {
```

Maybe return an error instead of just logging it?